v1.1/create-security-certificates-openssl.md

```diff
@@ -0,0 +1,292 @@
+
+## Examples
+
+### Generate the CA key and certificate pair
+
+1. Create two directories:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ mkdir certs
+    ~~~
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ mkdir my-safe-directory
+    ~~~
+    - `certs`: You'll generate your CA certificate and all node and client certificates and keys in this directory and then upload some of the files to your nodes.
+    - `my-safe-directory`: You'll generate your CA key in this directory and then reference the key when generating node and client certificates. After that, you'll keep the key safe and secret; you will not upload it to your nodes.
+
+2. Create the ca.conf file
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    [ ca ]
+    default_ca = CA_default
+
+    [ CA_default ]
+    default_days = 3660
+    database = index.txt
+    serial = serial.txt 
+    default_md = sha256 
+    copy_extensions = copy
+
+    # Used to generate the CA certificate.
+    [ req ]
+    prompt=no
+    distinguished_name = distinguished_name
+    x509_extensions = extensions
+
+    [ distinguished_name ]
+    organizationName = Cockroach
+    commonName = Cockroach CA
+
+    [ extensions ]
+    keyUsage = critical,digitalSignature,nonRepudiation,keyEncipherment,keyCertSign
+    basicConstraints = critical,CA:true,pathlen:1
+
+    # Common policy for nodes and users.
+    [ signing_policy ]
+    organizationName = supplied
+    commonName = supplied
+
+    # Used to sign node certificates.
+    [ signing_node_req ]
+    keyUsage = critical,digitalSignature,keyEncipherment
+    extendedKeyUsage = serverAuth,clientAuth
+
+    # Used to sign client certificates.
+    [ signing_client_req ]
+    keyUsage = critical,digitalSignature,keyEncipherment
+    extendedKeyUsage = clientAuth
+    ~~~
+
+3. Generate the CA key using the [`openssl genrsa`](https://wiki.openssl.org/index.php/Manual:Genrsa(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    openssl genrsa -out ${my-safe-directory}/ca.key 2048 2> /dev/null
+    chmod 400 ${my-safe-directory}/ca.key
+    ~~~
+
+4. Generate the CA certificate using the [`openssl req`](https://wiki.openssl.org/index.php/Manual:Req(1)) command:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    # Create CA certificate.
+    openssl req \
+      -new \
+      -x509 \
+      -config ca.conf \
+      -key ${CCERTS}/ca.key \
+      -out ${OSCERTS}/ca.crt \
+      -days 3660 \
+      -batch
+
+    # Reset database and index files.
+    rm -f index.txt serial.txt
+    touch index.txt
+    echo '01' > serial.txt
+    ~~~
+
+### Generate the certificate and key pairs for nodes
+
+1. Create the node.conf file for the first node:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    [ req ]
+    prompt=no
+    distinguished_name = distinguished_name
+    req_extensions = extensions
+
+    [ distinguished_name ]
+    organizationName = Cockroach
+    commonName = <node-hostname>
```

Comment:
this needs to be `node`, as in that actual string, not the hostname of the machine. When used as a client certificate, this is the username used by the various nodes.

Reply:
On reading about commonName on Google, I still think it should be the hostname of the node. Can you clarify please?

Reply:
"node"

Reply:
Also, "do not change this"